crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -499,6 +500,32 @@ trait SqliteObjectCryptoStoreExt: SqliteObjectExt {
             .await?)
     }
 
+    async fn mark_inbound_group_sessions_as_backed_up(&self, session_ids: Vec<Key>) -> Result<()> {
+        let max_chunk_size = usize::try_from(self.limit(Limit::SQLITE_LIMIT_VARIABLE_NUMBER).await)
+            .expect("SQLITE_LIMIT_VARIABLE_NUMBER was not a usize!");
```

Comment:
Do you know [`chunk_large_query_over`](https://github.com/matrix-org/matrix-rust-sdk/blob/18b6387a7ad8d24ca258a4ec92d6d4f7f20fa4e5/crates/matrix-sdk-sqlite/src/state_store.rs#L1608-L1652) exists? Maybe you can use it?

In case you can't, be aware that `Limit::SQLITE_LIMIT_VARIABLE_NUMBER` might to large if you add other “non-dynamic” placeholders in your query. I reckon it would be safer and more future-proof to subtract 10 to the limit, or —as an alternative—, add a comment around the `UPDATE` query saying to keep an eye to the `max_chunk_size`.

Reply:
Yep, I like your strategy of halving it.